src/builder.ts

```diff
@@ -96,6 +96,13 @@ export class UnknownQuery extends EntityQuery<z.ZodUnknown> {
     });
   }
 
+  filterByType(filterTypeValue: string) {
+    this.query += `[_type == '${filterTypeValue}']`;
+    return new UnknownArrayQuery({
+      ...this.value(),
+    });
+  }
```

Comment:
```suggestion
  filterByType(filterTypeValue: string) {
    return this.filter(`_type == '${filterTypeValue}'`);
  }
```

Just so we don't end up having to duplicate logic with `.filter`, perhaps we just use `this.filter` in this method.